File: src/main/java/com/kishlaly/ta/model/indicators/Bollinger.java
package com.kishlaly.ta.model.indicators;

import com.kishlaly.ta.model.EntityWithDate;

public class Bollinger extends EntityWithDate {

    private double bottom;
    private double middle;
    private double top;

    public Bollinger(final Long timestamp, final double bottom, final double middle, final double top) {
        super(timestamp);
        this.bottom = bottom;
        this.middle = middle;
        this.top = top;
    }

    public Long getTimestamp() {
        return this.timestamp;
    }

    public void setTimestamp(final Long timestamp) {
        this.timestamp = timestamp;
    }

    public double getBottom() {
        return this.bottom;
    }

    public void setBottom(final double bottom) {
        this.bottom = bottom;
    }

    public double getMiddle() {
        return this.middle;
    }

    public void setMiddle(final double middle) {
        this.middle = middle;
    }

    public double getTop() {
        return this.top;
    }

    public void setTop(final double top) {
        this.top = top;
    }
}

File: src/main/kotlin/com/kishlaly/ta/model/indicators/Bollinger.kt
package com.kishlaly.ta.model.indicators

import com.kishlaly.ta.model.AbstractModel

class Bollinger(
    timestamp: Long,
    val bottom: Double,
    val middle: Double,
    val top: Double
) : AbstractModel(timestamp) {

    override fun valuesPresent() = bottom != Double.NaN && middle != Double.NaN && top != Double.NaN

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Bollinger

        if (bottom != other.bottom) return false
        if (middle != other.middle) return false
        if (top != other.top) return false

        return true
    }

    override fun hashCode(): Int {
        var result = bottom.hashCode()
        result = 31 * result + middle.hashCode()
        result = 31 * result + top.hashCode()
        return result
    }


}